compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1813,7 +1813,7 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                 err.span_suggestions(
                     span,
                     msg,
-                    suggestable_variants.drain(..),
+                    suggestable_variants.clone().into_iter(),
```

Comment:
Instead of cloning the vector, can we cache the result of `is_empty` before this branch?
Still, replacing `drain` by `into_iter` is a great idea.

Reply:
Done